packages/@tailwindcss-upgrade/src/index.ts

```diff
@@ -142,27 +189,20 @@ async function run() {
       // Template migrations
 
       info('Migrating templates using the provided configuration file.')
-      for (let config of configBySheet.values()) {
-        let set = new Set<string>()
-        for (let globEntry of config.globs.flatMap((entry) => hoistStaticGlobParts(entry))) {
-          let files = await globby([globEntry.pattern], {
-            absolute: true,
-            gitignore: true,
-            cwd: globEntry.base,
-          })
-
-          for (let file of files) {
-            set.add(file)
+      for (let [stylesheet, candidatesByFile] of candidatesByFileBySheet.entries()) {
+        let config = configBySheet.get(stylesheet)!
+        for (let [file, { candidates, contents }] of candidatesByFile.entries()) {
+          let newContents = await migrateTemplateContents(
+            candidates,
+            config.designSystem,
+            config.userConfig,
+            contents,
+          )
+
+          if (newContents !== contents) {
+            await fs.writeFile(file, newContents)
```

Comment:
Only thing to note here is this is no longer happening in parallel. Probably not a big deal but at least wanted to point it out.

Reply:
Oh you're right, I only made the file reads parallel. Should do it here too 👍